server/routes/api/auth.js

```diff
@@ -0,0 +1,25 @@
+const express = require('express');
+const router = express.Router();
+const bcrypt = require('bcrypt');
+
+// User Model
+const User = require('../../models/user');
+
+// POST User Login
+router.post('/login', (req, res) => {
+  User.authenticate(req.body.email, req.body.password)
+    .then(
+      user => {
+        req.session.userId = user._id;
+        res.status(200).json(user);
+      },
+      err => {
+        res.status(500).json(err);
```

Comment:
500 is usually a server problem. Let's return a 400 here. Also, we're almost mimicking that basic structure. We'll want to place the error object in a property of 'error', so what will be returned is:

```
{
    "error": {
      "message": "Incorrect username and password combination.",
      "name": "AuthenticationError"
    }
}

Reply:
Changed error status to 400 and placed error object in "error" property.